lib/Doctrine/ORM/Mapping/EscapingQuoteStrategy.php

```diff
@@ -0,0 +1,176 @@
+    {
+        if(isset($joinColumn['quoted'])){
+            return $platform->quoteIdentifier($joinColumn['referencedColumnName']);
+        }
+        $reservedKeyList = $platform->getReservedKeywordsList();
+        if($reservedKeyList->isKeyword($joinColumn['referencedColumnName'])){
+            return $platform->quoteIdentifier($joinColumn['referencedColumnName']);
+        }
+        return $joinColumn['referencedColumnName'];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getJoinTableName(array $association, ClassMetadata $class, AbstractPlatform $platform)
+    {
+        if(isset($association['joinTable']['quoted'])){
+            return $platform->quoteIdentifier($association['joinTable']['name']);
+        }
+        $reservedKeyList = $platform->getReservedKeywordsList();
+        if($reservedKeyList->isKeyword($association['joinTable']['name'])){
+            return $platform->quoteIdentifier($association['joinTable']['name']);
+        }
+        return $association['joinTable']['name'];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getIdentifierColumnNames(ClassMetadata $class, AbstractPlatform $platform)
+    {
+        $quotedColumnNames = array();
+
+        foreach ($class->identifier as $fieldName) {
+            if (isset($class->fieldMappings[$fieldName])) {
+                $quotedColumnNames[] = $this->getColumnName($fieldName, $class, $platform);
+
+                continue;
+            }
+
+            // Association defined as Id field
+            $joinColumns            = $class->associationMappings[$fieldName]['joinColumns'];
+            $assocQuotedColumnNames = array_map(
+                function ($joinColumn) use ($platform)
+                {
+                    if(isset($joinColumn['quoted'])){
+                        return $platform->quoteIdentifier($joinColumn['name']);
+                    }
+                    $reservedKeyList = $platform->getReservedKeywordsList();
+                    if($reservedKeyList->isKeyword($joinColumn['name'])){
+                        return $platform->quoteIdentifier($joinColumn['name']);
+                    }
+                    return $joinColumn['name'];
+                },
+                $joinColumns
+            );
+
+            $quotedColumnNames = array_merge($quotedColumnNames, $assocQuotedColumnNames);
+        }
+
+        return $quotedColumnNames;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getColumnAlias($columnName, $counter, AbstractPlatform $platform, ClassMetadata $class = null)
+    {
+        // 1 ) Concatenate column name and counter
+        // 2 ) Trim the column alias to the maximum identifier length of the platform.
+        //     If the alias is to long, characters are cut off from the beginning.
+        // 3 ) Strip non alphanumeric characters
+        // 4 ) Prefix with "_" if the result its numeric
+        $columnName = $columnName . '_' . $counter;
```

Comment:
Don't overwrite the same variable: if the semantic meaning changes, then the variable should also have a different name